sdk/core/Azure.Core/src/TokenCredential.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Represents a credential capable of providing an OAuth token
+    /// </summary>
+    public abstract class TokenCredential
+    {
+        public abstract ValueTask<string> GetTokenAsync(string[] scopes, CancellationToken cancellationToken);
```

Comment:
I changed parameter type to `string[]` as it feels like something that would be enumerated very frequently and we'd like to avoid enumerator allocation per call